corehq/apps/app_manager/models.py

```diff
@@ -3987,6 +3988,8 @@ class ApplicationBase(VersionedDoc, SnapshotMixin,
     # always false for RemoteApp
     case_sharing = BooleanProperty(default=False)
 
+    language_profiles = DictProperty()
```

Comment:
I'd shy away from storing this as a dict of user-inputted-slugs to list of language codes. I think what the user types there should be treated as a display name and not as a "key" to the profile; it's natural to want to store this key as a reference to the profile, and then when the user changes the profile's display name, that reference becomes broken.

Instead, I'd suggest generating a unique key for each one, and having it be a `DictPropery(ApplicationProfile)`, where `ApplicationProfile` has `langs = StringListProperty()` and potentially other fields eventually. Alternatively, it could be a `ListProperty(ApplicationProfile)` and have `ApplicationProfile` include `key = StringProperty()` or some such thing.